### PR TITLE
Add URL rewrite for static file signature

### PR DESCRIPTION
**Pull request by mrkschan**

### CODE REVIEW

File: .magento.app.yaml [hunk 1]
```diff
@@ -55,6 +55,9 @@ web:
             allow: true
             scripts: false
             passthru: "/front-static.php"
```
Comment (damz) [inline .magento.app.yaml]: That file doesn't exist, right? It's some pretty old attempt at supporting static assets in development mode. Just remove that line.

File: .magento.app.yaml [hunk 2]
```diff
@@ -55,6 +55,9 @@ web:
...
+            rules:
+                /static/(version\d*/)?(?<resource>.*)$:
+                    passthru: "/static.php?resource=$resource"
```
Comment (damz) [inline .magento.app.yaml]: This should probably be:

```yaml
^/static/version\d+/(?<resource>.*)$:
  passthru: "/static/$resource"
```

The original from `nginx.conf.sample` is:

```
location ~ ^/static/version {
    rewrite ^/static/(version\d*/)?(.*)$ /static/$2 last;
}
```

But that regular expression doesn't make sense: (1) `\d*` should be `\d+` as you actually want a number in there and (2) the version group should not be optional.

That other part of the configuration:

```
if (!-f $request_filename) {
    rewrite ^/static/(version\d*/)?(.*)$ /static.php?resource=$2 last;
}
```

Is to support lazy static asset generation in development mode, we have no use for that (and given that the `passthru` above has been pointing to a non-existent file all this time, it sounds like nobody currently cares).

File: .magento.app.yaml (hunk 1 above)
Comment (mrkschan) [inline .magento.app.yaml]: that's the new static.php, right?

File: .magento.app.yaml (hunk 2 above)
Comment (mrkschan) [inline .magento.app.yaml]: so, we let it fallback to /static/normal-file-path which may fallback to static.php in passthru of /static, correct?

See https://github.com/magento/magento-cloud/pull/18#discussion_r88621879

File: .magento.app.yaml (hunk 1 above)
Comment (mrkschan) [inline .magento.app.yaml]: also check https://github.com/magento/magento-cloud/pull/16

**Comment by mazhalai** [comment]
Any update on this?  http://test-2-0-11-nxlg7rq-t6slh7zuphmei.us.magentosite.cloud/ is broken

**Comment by bwaters** [comment]
front-static.php is there mcc copies static.php to front-static.php.  This is  to work around "some platform.sh limitation" with /static.php and /static directory.  I am guessing based on your comments @damz  that no such limitation exists and we can just drop front-static.php and use /static just like on all of our other M2 installations.

**Pull request closed without merging**